test/t8_schemes/t8_gtest_nca.cxx

```diff
@@ -275,10 +290,15 @@ TEST_P(nca, resursive_check_higher_level)
         num_children = ts->t8_element_num_children(correct_nca_high_level);
         if(num_children > 1)
```

Comment:
Please improve code formatting and add a comment what you are exactly doing within this if-clause. Are you checking the children with each other in the two nested loops? If so, why not test all children with all siblings, even with itself? Also with switched parameter passing: `check(childA, childB)` and `check(childB, childA)`. No need to optimized the code here. ;)

Reply:
Added a comment and switched parameter passing! This discovered a bug, which I fixed. The children are not checked within the nexted loops, and we shouldn't check the children with each other here. To enforce that the nca of childA and childB is 'correct_nca' they have to be different, such that they and their children are always on differen subtrees that only have 'correct_nca' as their nearest common ancestor.